indicators/qc.py

```diff
@@ -0,0 +1,125 @@
+"""Script for performing QC checks on files produced via indicators.py and their individual slurm job output files.
+This script uses the {out_dir}/qc/qc.csv file produced in the indicators/slurm.py script as a "to-do list" of files to check.
+QC errors are written to {out_dir}/qc/qc_error.txt. 
+
+Usage:
+    python qc.py --out_dir /beegfs/CMIP6/jdpaul3/scratch
+"""
+
+import argparse
+import pandas as pd
+import os
+import xarray as xr
+import numpy as np
+from pathlib import Path
+from luts import units_lu, ranges_lu
+
+
+def qc_by_row(row, error_file):
+
+    # set up list to collect error strings
+    error_strings = []
+
+    # QC 1: does the slurm job output file exist? And does it show success message?
+    if os.path.isfile(row[2]) == False:
+        error_strings.append(f"ERROR: Expected job output file {row[2]} not found.")
+    else:
+        with open(row[2], "r") as o:
+            lines = o.read().splitlines()
+            if len(lines) > 0:
+                if not lines[-1] == 'Job Completed':
+                    error_strings.append(f"ERROR: Slurm job not completed. See {row[2]}.")
+            else:
+                error_strings.append(f"ERROR: Slurm job output is empty. See {row[2]}.")
+
+    # QC 2: does the indicator .nc file exist?
+    if os.path.isfile(row[1]) == False:
+        error_strings.append(f"ERROR: Expected indicator file {row[1]} not found.")
+
+    # QC 3: do the indicator string, indicator .nc filename, and indicator variable name in dataset match?
+    qc_indicator_string = row[0]
+    fp = Path(row[1])
+    fp_indicator_string = fp.parts[-1].split("_")[0]
+
+    try:  # also checks that the dataset opens
+        ds = xr.open_dataset(fp)
+        ds_indicator_string = list(ds.data_vars)[0]
+    except:
+        error_strings.append(f"ERROR: Could not open dataset: {row[1]}.")
+        ds_indicator_string = "ERROR"
+
+    if not fp_indicator_string == ds_indicator_string == qc_indicator_string:
+        error_strings.append(
+            f"ERROR: Mismatch of indicator strings found between parameter: {qc_indicator_string}, filename: {row[1]}, and dataset variable: {ds_indicator_string}."
```

Comment:
I can't figure out what is going on in my output file in my test error case, but one instance of this message not being separated with the newline as expected, and looks like

```
ERROR: Mismatch of indicator strings found between parameter: rx1day, filename: /import/beegfs/CMIP6/kmredilla/indicators/output/historical/rx1day/rx1day__historical_indicator.nc, and dataset variable: ERROR.ERROR: Expected indicator file /import/beegfs/CMIP6/kmredilla/indicators/output/ssp585/rx1day/rx1day__ssp585_indicator.nc not found.
```

instead of 

```
ERROR: Mismatch of indicator strings found between parameter: rx1day, filename: /import/beegfs/CMIP6/kmredilla/indicators/output/historical/rx1day/rx1day__historical_indicator.nc, and dataset variable: ERROR.
ERROR: Expected indicator file /import/beegfs/CMIP6/kmredilla/indicators/output/ssp585/rx1day/rx1day__ssp585_indicator.nc not found.
```

Not sure what's going on and it's not a big thing, just wanted to point it out!

Reply:
Weird, I will look into this!